kpagacz/wiki-on-fire
# devops: added the AWS lambda to download data from wikimedia

* downloads the data every day
* stores in the dynamodb table
* created a github workflow to udpate the lambda every time the lambda directory is updated on the main branch
* modified the TestLambda.yaml cloudformation yaml to accommodate the real wikimedia lambda
Closes #138

How to test:
* there are multiple parts to this: the javascript code itself, the cloudformation lambda file, the github workflow
* the expected course of action to setup this is:
1. Create a DynamoDB table (can be done via the cloudformation file in: https://github.com/kpagacz/wiki-on-fire/pull/182)
2. Setup the lambda and roles and events (using the cloudformation yaml in this PR)
3. Test the lambda via the Amazon console (there is a UI for this)
![image](https://user-images.githubusercontent.com/24480246/147791760-12018bb1-9ccf-4ed3-aefa-be20900880d4.png)
The expected output should be a successful execution.
To test the lambda itself, the easiest is to upload the zipped lambda directory to aws lambda service. steps to do this:
  -  zip the contents of the directory: `zip -r function.zip .`
  -  upload the archive:
![image](https://user-images.githubusercontent.com/24480246/147791842-8969a005-1d83-4e61-802f-855f55106f72.png)
4. Test the workflow - it should be possible to trigger it manually. It should upload the zipped archive to the AWS automatically. You can check that the upload was successful by looking at the last modified field in the aws console:
![image](https://user-images.githubusercontent.com/24480246/147791875-09c655d2-c3b8-408c-b079-374666703283.png)
5. Test that the event triggers the lambda - I would advise to create a similar trigger that trigger every minute or similar, so you can observe the execution in a sensible time frame. You can add triggers here in the aws console:
![image](https://user-images.githubusercontent.com/24480246/147792018-3ad77829-a33f-444f-b985-3ee51ac8877d.png)
